src/ip.rs

```diff
@@ -0,0 +1,117 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+/// The kinds of networks that can be simulated in turmoil
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub enum IpVersion {
+    /// An Ipv4 network with an address space of 192.168.0.0/16
+    #[default]
+    V4,
+    /// An local area Ipv6 network with an address space of fe80::/64
+    V6,
+}
+
+impl IpVersion {
+    pub(crate) fn iter(&self) -> IpVersionAddrIter {
+        match self {
+            Self::V4 => IpVersionAddrIter::V4(1),
+            Self::V6 => IpVersionAddrIter::V6(1),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
nit: I don't think you want to copy or clone (or probably any of these derives) this as you'd have different next return values across each instance.